### PR TITLE
Feature/99 alert mechanism

### LINKED ISSUE #99: Add alert messages management mechanism to the UI
## Overview of the Feature Request
We need an alert message mechanism in the UI to show the different alert messages to the user.

These are some examples of the Alert messages

image

## What kind of user is the feature intended for?

Default user role since we are not handling any user role at the moment

## What inspired the request?
We are implementing the Dataset page and we need to show different alert messages for each state. 

## What existing behavior do you want changed?
None

## Any brand new behavior do you want to add to Dataverse?
Alert messages management

## Any open or closed issues related to this feature request?
- #58

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
This PR adds logic to the JSDatasetMapper that will create DatasetAlert objects to be used in the Dataset View page
## Which issue(s) this PR closes:
 
- Closes #99 

## Suggestions on how to test this:
Run the Cypress JSDatasetMapper unit test

### CODE REVIEW

File: tests/component/sections/dataset/dataset-files/files-table/files-info/file-info-cell/file-info-data/FileDate.spec.tsx [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+    const fileDate = new Date('2023-09-18')
+    const date = { type: FileDateType.PUBLISHED, date: fileDate }
     cy.customMount(<FileDate date={date} />)
-
-    cy.findByText(`Published Sep 18, 2023`).should('exist')
+    const dateString = fileDate.toLocaleDateString(Intl.DateTimeFormat().resolvedOptions().locale, {
+      year: 'numeric',
+      month: 'short',
+      day: 'numeric'
+    })
```
Comment (MellyGray) [inline tests/component/sections/dataset/dataset-files/files-table/files-info/file-info-cell/file-info-data/FileDate.spec.tsx:13]: Since this dateString is depending on the user's locale, this and the other changed tests are failing for me, because the FileDate component is implemented to always show the US date format and the test is checking the user's locale format, which in my case is the Spanish format

So either we change the test to check for the US date format or we refactor the FileDate component to show the user's date format.

In JSF I think it shows always the US format so that's why I implemented this like that, but I think that changing it to the user's locale could be an improvement, should we discuss this with Jullian?

File: src/dataset/infrastructure/mappers/JSDatasetMapper.ts [hunk 2]
```diff
@@ -29,19 +35,22 @@ export class JSDatasetMapper {
...
+      ),
+      privateUrl
     ).build()
   }
 
   static toVersion(
     jDatasetVersionId: number,
-    jsDatasetVersionInfo: JSDatasetVersionInfo
+    jsDatasetVersionInfo: JSDatasetVersionInfo,
+    requestedVersion?: string
```
Comment (MellyGray) [inline src/dataset/infrastructure/mappers/JSDatasetMapper.ts:46]: There are some e2e tests that are checking the DatasetVersion that need to be fixed because of the added requestedVersion property

**Review by MellyGray — COMMENTED** [review]
I really like the implementation of the Dataset Alerts as part of the Dataset model 🎉 I just left a few minor comments

I was also thinking about the other alerts, the ones that are not related with the Dataset Model but with certain behaviours of the user, such as 'there was an error when you clicked some button' Are we planning to implement those in the [ other alerts issue](https://github.com/IQSS/dataverse-frontend/issues/127) or should we create a new issue for this type of alerts not related with domain models?

File: tests/component/sections/dataset/dataset-files/files-table/files-info/file-info-cell/file-info-data/FileDate.spec.tsx (hunk 1 above)
Comment (ekraffmiller) [reply tests/component/sections/dataset/dataset-files/files-table/files-info/file-info-cell/file-info-data/FileDate.spec.tsx:13]: Ok, I see what you mean about locale format, that makes sense.  I think it's a good idea to use the user's  locale format.  I made the change because my local time zone was showing a different day since I'm five hours ahead, but that is fixed by testing against a dynamic date string, rather than changing the locale.
I can put a message in dv-tech to confirm about using the user's locale format, since it's different from the JSF

**Comment by ekraffmiller** [comment]
Hi Melina, I think I addressed all your change requests - the e2e test is failing, but in a different way on the github action than in my local environment - maybe that's because they are including different dataverse images?

**Comment by MellyGray** [comment]
> Hi Melina, I think I addressed all your change requests - the e2e test is failing, but in a different way on the github action than in my local environment - maybe that's because they are including different dataverse images?

@ekraffmiller the FileEmbargoDate component tests are still failing for me
image

Regarding the e2e tests running in the Github action, I had the same question and Guillermo told me that the Github Action is pointing to the dataverse develop branch image because ideally the backend changes should be merged before we start working on the frontend. But that's not the case right now, so some tests are failing in the GitHub action. 

But apparently it's ok, now the backend PR's have been given priority, so hopefully the backend will be ahead of the frontend soon

So I ran the e2e locally and I'm having this error, but I already fixed it in another branch:

image

Then some e2e tests fail from time to time because of the wait(), Cypress doesn't like being waited for so long, sometimes it crashes, we'll have to address that at some point but for this PR, I think it's ok like this.

In summary, can you take a look at the unit tests for the FileEmbargoDate component? the e2e test are ok like this

**Comment by ekraffmiller** [comment]
Hi Melina, Gustavo suggested I create a spike for the question of the dates being returned by the API, so I created this issue https://github.com/IQSS/dataverse/issues/9979.  I think the locale logic is separate from the timezone issue, and we have agreement on using the user's locale, so I don't think this PR is dependent on that, but let me know what you think.

**Review by MellyGray — APPROVED** [review]
LGTM 😄
